lms/djangoapps/course_api/blocks/permissions.py

```diff
@@ -39,15 +40,14 @@ def can_access_self_blocks(requesting_user: User, course_key: CourseKey) -> Acce
     )
     if user_is_enrolled_or_staff:
         return ACCESS_GRANTED
-    try:
-        return is_course_public(course_key)
-    except ValueError:
-        return ACCESS_DENIED
+    return is_course_public(course_key)
 
 
 def is_course_public(course_key: CourseKey) -> AccessResponse:
     """
     This checks if a course is publicly accessible or not.
     """
+    if not CourseOverview.course_exists(course_key):
+        return ACCESS_DENIED
```

Comment:
We do 403 instead of 404 in order to avoid leaking potentially private information of whether or not a course exists to unauthorized users.

Reply:
When I feel the need to add a comment to my own PR, I often ask myself if it should instead be a code comment. The answer is not always yes, but I’m this case, I think the comment is a reasonable one.